cluster-autoscaler/cloudprovider/mcm/mcm_cloud_provider.go

```diff
@@ -539,12 +567,17 @@ func buildMachineDeploymentFromSpec(value string, mcmManager *McmManager) (*Mach
 	return machinedeployment, nil
 }
 
+func getMachinesMarkedByCAForDeletion(mcd *v1alpha1.MachineDeployment) sets.Set[string] {
```

Comment:
Perhaps using the "comma, ok" idiom here for `mcd.Annotations[machinesMarkedByCAForDeletion]` would be good  for safety to avoid any case where a panic is issued when annotation is not found.

Reply:
Isn't it allowed to read from a nil map? Only writes will cause panic, right?

Reply:
`getMachinesMarkedByCAForDeletion`->`getMachineNamesMarkedByCAForDeletion`